huggingface/pytorch/training/docker/2.1/py3/cu121/Dockerfile.gpu

```diff
@@ -20,6 +20,8 @@ ARG MULTIPROCESS_VERSION=0.70.16
 ARG DILL_VERSION=0.3.8
 ARG NINJA_VERSION=1.11.1
 ARG PYTHON=python3
+ARG SMD_MODEL_PARALLEL_URL=https://sagemaker-distributed-model-parallel.s3.us-west-2.amazonaws.com/pytorch-2.0.0/build-artifacts/2023-04-14-20-14/smdistributed_modelparallel-1.15.0-cp310-cp310-linux_x86_64.whl
```

Comment:
1. Why is this model parallel dependency needed?

2. sanity check: Its defaulting to us-west-2 region endpoint. Will this cause any issue?

Reply:
1. This is the main error message I am seeing in CodeBuild failures: `ModuleNotFoundError: No module named 'smdistributed.modelparallel'`. This is occurring in Training Script. https://tiny.amazon.com/vldsc2wq/IsenLink. I do not have 100% confidence this is the fix, but I have seen other images add this dependency as well, so i think it is worth a shot.

2. I do not believe so, once again I am seeing this in other image dependencies as well. I can send this PR to DLC team member, see if they have more context.

Reply:
Approving the change. But lets get this to DLC team to review before merging.